ironfist/src/cpp/combat/creatures.cpp

```diff
@@ -137,6 +150,31 @@ void LoadCreatures() {
 						}
 				}
 
+				for (int i = 0; i < 6; i++) {
+					gMonSecondaryCost[id][i][0] = 0;
+					gMonSecondaryCost[id][i][1] = 0;
+				}
+				bool secondary_cost = false;
+				for (creature_t::secondary_cost_iterator i = c.secondary_cost().begin();
+				     i != c.secondary_cost().end();
+					 ++i) {
+					    for (int k = 0; k < ELEMENTS_IN(secondaryCostIdxTable); k++) {
+						   if (strcmp(secondaryCostIdxTable[k].resource, i->resource().c_str())
+							   == 0) {
+							   if (!secondary_cost) {
+								   gMonSecondaryCost[id][k][0] = 1;

```

Comment:
gMonSecondary cost should probably just have dimension [MAX_CREATURES][NUM_SECONDARY_RESOURCES]. That 0/1 field is redundant (I'll explain below how to eliminate the only place it's used).

And yeah, having some constant somewhere like "NUM_SECONDARY_RESOURCES" is much preferred to having the magic number "6" floating around.